### DISCUSSION TITLE
AIS-Catcher on Raspberry Pi Zero 2W

### DISCUSSION THREAD

**Discussion by jonkan81** [jvde-github/AIS-catcher#279] [Q&A] [answered]
I am trying to setup AIS-catcher on my boat using an RTL-SDR v4 dongle and a Raspberry Pi Zero 2W. I tested the dongle and configuration out first with my laptop and got it working at the first try. Got 4-5 ais targets and managed to forward the data to SignalK as I intended to. Then I moved the dongle to my Pi and used the same configuration (I added the -F flag as the documentation said I should) and started it up. No targets at all. I have tried to lower the sample rate to 288K and I have tried a lot of different configurations on the dongle, but nothing seems to work.

The logs looks pretty similar as on the laptop.

Has anyone else got any experience with this and how to troubleshoot it? I don't want to have my laptop on the boat the whole summer :-(

**Comment by jvde-github** [comment]
One possible explanation is that the special drivers for the V4 are not installed on the Zero? What do you see as output of the program, any references it has picked up it is a V4?

**Reply by jonkan81** [reply]
This is what I get:

May 23 02:14:44 aisreceiver aiscatcher[561]: This is free software; see the source for copying conditions.There is NO
May 23 02:14:44 aisreceiver aiscatcher[561]: warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
May 23 02:14:44 aisreceiver aiscatcher[561]: Adding plugin (/usr/share/aiscatcher/my-plugins/shipxplorer_option.pjs). Description: "add option to look up vessel details in ShipXplorer", Author: "jvde-github", version 2
May 23 02:14:44 aisreceiver aiscatcher[561]: Adding plugin (/usr/share/aiscatcher/my-plugins/vesselfinder_option.pjs). Description: "add option to look up vessel details in VesselFinder", Author: "jvde-github", version 2
May 23 02:14:44 aisreceiver aiscatcher[561]: Adding plugin (/usr/share/aiscatcher/my-plugins/marinetraffic_option.pjs). Description: "add option to look up vessel details in MarineTraffic", Author: "jvde-github", version 2
May 23 02:14:44 aisreceiver aiscatcher[561]: Adding plugin (/usr/share/aiscatcher/my-plugins/google_option.pjs). Description: "add option to look up location in Google Maps", Author: "jvde-github", version 2
May 23 02:14:44 aisreceiver aiscatcher[561]: Searching for device with SN 00000162.
May 23 02:14:44 aisreceiver aiscatcher[561]: Detached kernel driver
May 23 02:14:44 aisreceiver aiscatcher[561]: Found Rafael Micro R828D tuner
May 23 02:14:45 aisreceiver aiscatcher[561]: UDP: open socket for host: 192.168.42.1, port: 11010, filter: OFF, JSON: OFF
May 23 02:14:45 aisreceiver aiscatcher[561]: TCP Server: start thread at port 8383
May 23 02:14:45 aisreceiver aiscatcher[561]: HTML Server running at port 8383
May 23 02:14:45 aisreceiver aiscatcher[561]: [R82XX] PLL not locked!
May 23 02:14:45 aisreceiver aiscatcher[561]: [R82XX] PLL not locked!
May 23 02:14:45 aisreceiver aiscatcher[561]: [R82XX] PLL not locked!
May 23 02:14:45 aisreceiver aiscatcher[561]: Allocating 15 zero-copy buffers
May 23 02:14:45 aisreceiver aiscatcher[561]: Device    : Blog V4
May 23 02:14:45 aisreceiver aiscatcher[561]: Settings  : rate 288K bw 192K format CU8 tuner AUTO rtlagc OFF biastee OFF buffer_count 24
May 23 02:14:45 aisreceiver aiscatcher[561]: Model #0 -> (Src: 1, Grp: 1): [AIS engine v0.59] ps_ema ON afc_wide ON droop ON fp_ds ON dsk OFF
May 23 02:14:55 aisreceiver aiscatcher[561]: [AIS engine v0.59 #0-0]                 received: 0 msgs, total: 0 msgs, rate: 0 msg/s

**Reply by jonkan81** [reply]
It looks like it has picked up it as the correct kind of device. The only thing that doesn't look entirely swell is the three rows with "[R82XX] PLL not locked!". I don't know what that means.

**Reply by jvde-github** [reply, marked as answer]
Just looked with my V4 and have the same output. For my reference did you compile the rtl-sdr driver from source for this device?

The point is that with the old driver, it seems that it is running, you just don't receive any data.

**Reply by jonkan81** [reply]
Thanks for the suggestion. That might be the issue. I don't remember how I installed the driver, so probably I did it the wrong way ;-)

I have compiled and reinstalled the driver now and will test it tomorrow when I'm in the boat again.

**Reply by jonkan81** [reply]
Updating the driver did the trick. Thank you for the help!
